node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java

```diff
@@ -53,28 +56,29 @@ public List<Autoscaling> suggest(Application application, Cluster cluster, NodeL
     /**
      * Autoscale a cluster by load. This returns a better allocation (if found) inside the min and max limits.
      *
-     * @param clusterNodes the list of all the active nodes in a cluster
-     * @param enableDetailedLogging Whether to log autoscaling decision data
+     * @param clusterNodes          the list of all the active nodes in a cluster
+     * @param enabled               Whether autoscaling is enabled
+     * @param logDetails            Whether to log decision details
      * @return scaling advice for this cluster
      */
-    public Autoscaling autoscale(Application application, Cluster cluster, NodeList clusterNodes, boolean enableDetailedLogging) {
+    public Autoscaling autoscale(Application application, Cluster cluster, NodeList clusterNodes, boolean enabled, boolean logDetails) {
         var limits = Limits.of(cluster);
         var model = model(application, cluster, clusterNodes);
         if (model.isEmpty()) return Autoscaling.empty();
 
-        if (! limits.isEmpty() && cluster.minResources().equals(cluster.maxResources()))
+        if (!enabled || (! limits.isEmpty() && cluster.minResources().equals(cluster.maxResources())))
             return Autoscaling.dontScale(Autoscaling.Status.unavailable, "Autoscaling is not enabled", model);
```

Comment:
Maybe we should output two different messages depending on the reason so that people get why it's disabled even if they themselves have enabled it?